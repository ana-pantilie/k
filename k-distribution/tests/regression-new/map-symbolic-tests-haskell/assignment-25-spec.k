// Copyright (c) 2019 K Team. All Rights Reserved.
// generated by gen-tests.sh

requires "map-tests.k"

module ASSIGNMENT-25-SPEC
    imports MAP-TESTS

    claim <k> assignment ( (X:MyId |-> 4) [ X <- 3 ] ) => assignmentResult ( (X:MyId |-> 3) ) </k>

endmodule
