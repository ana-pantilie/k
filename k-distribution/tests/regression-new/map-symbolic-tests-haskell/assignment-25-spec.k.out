  #Not ( {
    X
  #Equals
    Y
  } )
#And
  #Not ( {
    X
  #Equals
    Z
  } )
#And
  #Not ( {
    Y
  #Equals
    Z
  } )
#And
  #Not ( {
    false
  #Equals
    Z:MyId in_keys ( MAP )
  } )
#And
  <k>
    assignment ( MAP
    X:MyId |-> 1
    Y:MyId |-> 2
    Z:MyId |-> 3 ) ~> .
  </k>
#And
  {
    false
  #Equals
    X:MyId in_keys ( MAP )
  }
#And
  {
    false
  #Equals
    Y:MyId in_keys ( MAP )
  }
